### PR TITLE
Fix disabling of ``bad-option-value``

### LINKED ISSUE #3312: Can't disable bad-option-value
### Steps to reproduce
1. Write code on a computer with a somewhat new pylint (2.4.3 in my example). Get a warning like `useless-object-inheritance` that I want to ignore, as I'm writing code compatible with python2 and python3.
2. Disable said warning with `# pylint: disable=useless-object-inheritance`.
3. Get a "Bad option value" when other people run their pylint version (example: 2.3.1; and by people, sometimes I mean docker instances ran from Jenkins that I would rather not rebuild or that depend on other people and I can't modify)
4. Try to disable said error with a global `# pylint: disable=bad-option-value`

### Current behavior
`# pylint: disable=bad-option-value` is ignored
`# pylint: disable=E0012` is ignored

### Expected behavior
To be able to write code that works on several versions of pylint and not having to make sure every computer in the company and every docker container has the same pylint version.

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #3312.

### CODE REVIEW

File: tests/functional/b/bad_option_value_disable.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+var = 1  # pylint: disable=a-removed-option
+
+# pylint: enable=bad-option-value
+
+var = 1  # pylint: disable=a-removed-option # [bad-option-value]
+
+# bad-option-value needs to be disabled before the bad option
+var = 1  # pylint: disable=a-removed-option, bad-option-value # [bad-option-value]
+var = 1  # pylint: disable=bad-option-value, a-removed-option
```
Comment (Pierre-Sassoulas) [inline tests/functional/b/bad_option_value_disable.py:14]: My first instinct was "nice", but then I don't think it's instinctive, maybe it's better if it disable whatever the order ? After all for all other messages disabling it disable for the line and the order is not important.

File: tests/regrtest_data/long_test_file.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+
+
+
+
+
+
+
+print(1)
```
Comment (Pierre-Sassoulas) [inline tests/regrtest_data/long_test_file.py:100]: Seems like this file is unrelated, right ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
Pylint 3.0 will be *SO* clean :)

File: tests/regrtest_data/long_test_file.py (hunk 2 above)
Comment (DanielNoord) [reply tests/regrtest_data/long_test_file.py:100]: No it is used for the test. I needed a module that isn’t restricted by line numbers as the enable tests are using various line numbers to test the behaviour.
Using an empty module creates issue as you can’t disable something on line 10 if it doesn’t exist. Thus, I created an extra long file so we never run into issue with that.

File: tests/regrtest_data/long_test_file.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+
```
Comment (Pierre-Sassoulas) [inline tests/regrtest_data/long_test_file.py:1]:
```suggestion
"""
This file is used for bad-option-value's test. We needed a module that isn’t restricted by line numbers
as the enable tests are using various line numbers to test the behaviour.

Using an empty module creates issue as you can’t disable something on line 10 if it doesn’t exist. Thus, we created an extra long file so we never run into issue with that.
"""
```

File: tests/regrtest_data/long_test_file.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+
+
+
+
```
Comment (Pierre-Sassoulas) [inline tests/regrtest_data/long_test_file.py:6]:
```suggestion
"""
This file is used for bad-option-value's test. We needed a module that isn’t restricted by line numbers
as the enable tests are using various line numbers to test the behaviour.

Using an empty module creates issue as you can’t disable something on line 10 if it doesn’t exist. Thus, we created an extra long file so we never run into issue with that.
"""
```

File: tests/functional/b/bad_option_value_disable.py (hunk 1 above)
Comment (DanielNoord) [reply tests/functional/b/bad_option_value_disable.py:14]: Hm, that would again be some additional code for very specific rare circumstances.

I think there is a very small group of users that will want to disable `bad-option-value` inline. Then there is an even smaller group who will make this mistake. I think that group is also likely to try to shuffle around the pragma's to fix the issue before opening an issue on the repo.
It's just a lot of additional code for something that I think can actually be "expected". Parsing pragma's in order is not something that is very out of the ordinary.

_My hesitance about this comes from the difficulty I had with this PR. You don't want to know how specific our current system is. Disable's on the first line counting for line zero but not for line two, disable's on function definitions also counting for decorators, etc. I'd rather simplify that make it more complicated again._

File: tests/regrtest_data/long_test_file.py (hunk 4 above)
Comment (DanielNoord) [reply tests/regrtest_data/long_test_file.py:6]: I'm going to commit this and see if tests pass. We use those first lines quite a bit in the test, so I'm not sure if those can be a doctoring.

File: tests/functional/b/bad_option_value_disable.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply tests/functional/b/bad_option_value_disable.py:14]: Fair enough

File: ChangeLog [hunk 5]
```diff
@@ -9,6 +9,12 @@ Release date: TBA
 ..
   Put new features here and also in 'doc/whatsnew/2.14.rst'
 
+* We have improved our recognition of inline disable and enable comments. It is
+  now possible to disable ``bad-option-value`` inline as well as certain other
+  previously unsupported messages.
```
Comment (Pierre-Sassoulas) [inline ChangeLog:14]:
```suggestion
* We have improved our recognition of inline disable and enable comments. It is
  now possible to disable ``bad-option-value`` inline  (as long as you disable it before
  the bad option value is raised, i.e. ``disable=bad-option-value,bad-message`` not ``disable=bad-message,bad-option-value`` ) as well as certain other previously unsupported messages.
```

File: doc/whatsnew/2.14.rst [hunk 6]
```diff
@@ -129,6 +129,12 @@ Other Changes
 
   Closes #4301
 
+* We have improved our recognition of inline disable and enable comments. It is
+  now possible to disable ``bad-option-value`` inline as well as certain other
+  previously unsupported messages.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2.14.rst:134]:
```suggestion
* We have improved our recognition of inline disable and enable comments. It is
  now possible to disable ``bad-option-value`` inline (as long as you disable it before
  the bad option value is raised, i.e. ``disable=bad-option-value,bad-message`` not ``disable=bad-message,bad-option-value`` ) as well as certain other
  previously unsupported messages.
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you @DanielNoord amazing work !

**Review by jacobtylerwalls — APPROVED** [review]
Thanks for taking this on. This really adds value to the beta.

I might track in a follow up issue (nothing blocking us now!) to allow `--disable=all` to disable the messages emitted for a `.pylintrc`, as it's a little unexpected to not be able to use that:

```python
% pylint music21 --disable=all 
************* Module /Users/.../music21/.pylintrc
.pylintrc:1:0: E0012: Bad option value for --disable. Don't recognize message bad-whitespace. (bad-option-value)
.pylintrc:1:0: E0012: Bad option value for --disable. Don't recognize message bad-continuation. (bad-option-value)
.pylintrc:1:0: E0012: Bad option value for --disable. Don't recognize message no-self-use. (bad-option-value)
.pylintrc:1:0: E0012: Bad option value for --disable. Don't recognize message star-args. (bad-option-value)
************* Module /Users/.../music21/.pylintrc
.pylintrc:1:0: E0015: Unrecognized option found: profile, files-output, bad-functions, function-name-hint, variable-name-hint, const-name-hint, attr-name-hint, argument-name-hint, class-attribute-name-hint, inlinevar-name-hint, class-name-hint, module-name-hint, method-name-hint, no-space-check (unrecognized-option)
```
